### PR TITLE
Optimize workout update

### PR DESCRIPTION
**Pull request by Hubertsci**

### CODE REVIEW

**Review by Ka1zerov — CHANGES_REQUESTED** [review]
Mainly, we should divide WorkoutService into WorkoutService, WorkoutExerciseService, WorkoutExerciseSetService. Same with mappers.
Then it would go along with Single Responsibility Principle. 
It will also make the code more modular and improve readability a lot.

**Comment by Ka1zerov** [comment]
updateEntity methods should not be in mappers. (if its an update let it be handled in service)

In WorkoutExerciseService:
updateWorkoutExercises should be refactored into several methods:
updateWorkoutExercises
updateExistingWorkoutExercise
addNewWorkoutExercises
findExerciseById

Same in WorkoutExerciseSetService

also, try using streams api instead of the iterator (it is really more readable)

Everything else seems to be really great (we will also need some tests for all of this)

File: src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java [hunk 1]
```diff
@@ -58,4 +58,14 @@ public void setExercises(List<WorkoutExercise> exercises) {
         }
```
Comment (Ka1zerov) [inline src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java:58]: remove this method (since we are not using in Update scenario @Setter annotation is enough)

**Review by Ka1zerov — CHANGES_REQUESTED** [review]

**Comment by Hubertsci** [comment]
> updateEntity methods should not be in mappers. (if its an update let it be handled in service)
> 
> In WorkoutExerciseService: updateWorkoutExercises should be refactored into several methods: updateWorkoutExercises updateExistingWorkoutExercise addNewWorkoutExercises findExerciseById
> 
> Same in WorkoutExerciseSetService (the general rule is that methods should not be too long, if those are, they should be refactored into several others, naming is a key factor here, and since we really need to understand what is happening - refactoring is great deed)
> 
> also, try using streams api instead of the iterator (it is really more readable)
> 
> Everything else seems to be really great (we will also need some tests for all of this)

Those are very nice hints to improve readability. I already refactored the code - decided to leave iterators, because IMO streams with too much logic become less readable.

File: src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java (hunk 1 above)
Comment (Hubertsci) [reply src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java:58]: But once again it's a proper logic of a setter for this kind of field with relationship, the one generated by lombok can cause us some troubles and generate errors.

**Comment by Ka1zerov** [comment]
LGTM!
now we need tests and then we can merge

File: src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java
Comment (Ka1zerov) [inline src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java:1]: We dont need a fetch type lazy here.
Its either Eager or we should use EntityGraphs in order to fetch all data

**Review by Ka1zerov — CHANGES_REQUESTED** [review]

File: src/main/java/com/sourcery/gymapp/backend/workout/service/WorkoutExerciseService.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            WorkoutExercise workoutExercise
+    ) {
+        return updateWorkoutDto.exercises().stream()
+                .filter(exerciseDto -> exerciseDto.id() != null && exerciseDto.id().equals(workoutExercise.getId()))
+                .findFirst()
+                .orElseThrow();
+    }
+
+    private Exercise findExerciseById(UUID id) {
+        return exerciseRepository.findById(id)
```
Comment (Ka1zerov) [inline src/main/java/com/sourcery/gymapp/backend/workout/service/WorkoutExerciseService.java:113]: it should call exercise service, not repository directly :C

**Review by Ka1zerov — CHANGES_REQUESTED** [review]
Ensure to test edge cases for critical scenarios:
- Handling of empty or null `exercises` lists during creation or update.
- Correct behavior for invalid or missing IDs in `CreateWorkoutExerciseDto` or `CreateWorkoutExerciseSetDto`.
- Boundary values for fields like `setNumber`, `reps`, `weight`, and `restTime`.
- Preservation of existing workout details when adding or removing exercises or sets.
These tests are essential to validate logic beyond annotation-driven constraints.

**Review by Ka1zerov — CHANGES_REQUESTED** [review]
One vital edge case to must-have:
testing CreateWorkoutDto and UpdateWorkoutDto where the same exercise (same exerciseId) appears several times in one training (at least twice) with different orderNumbers. This will ensure the logic handles duplicate entries correctly based on order

**Review by Ka1zerov — APPROVED** [review]
🎉🎊🎈🥳✨🎂🎁🎆🍾🎇🎉🎊🎈🍸🎆🎁🎉
                                   _**### LGTM!!!**_
🎊🎈🎉🍾🎁🎇🎆✨🥂🎂🎉🎊🎈🍸🎇🎁✨
